### ISSUE TITLE
pretrained weights on google colab

### ISSUE THREAD

**Issue by Salmankm93** [ultralytics/ultralytics#20326]
Hi everyone;
I have used Google Colab to train the yolo. However, after cloning the repo, I always train yolo from scratch using the .yaml file. However, when I set the sources for CPU, the pretrained weights don't download, but when I change to T4 suddenly it downloads the pretrained weights and starts to use it, which I don't want So my main question is how to train from scratch using a " .yaml" file and with "T4" resource on Google Colab, and thanks.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Salmankm93, thank you for reaching out and for using Ultralytics 🚀! This is an automated response to help guide you while an Ultralytics engineer reviews your issue and will assist you soon.

We recommend checking out the [Docs](https://docs.ultralytics.com/) for guidance on training workflows, including [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples. Many common questions are addressed there.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) so we can better understand and debug your issue.

For custom training ❓ questions like this one, please include as much detail as possible, such as your training command, config, and logs, to help us assist you efficiently. Also, review our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) to ensure you're following best practices.

Join the Ultralytics community where you prefer:  
- Real-time chat: [Discord](https://discord.com/invite/ultralytics) 🎧  
- In-depth discussions: [Discourse](https://community.ultralytics.com/)  
- Community threads: [Subreddit](https://reddit.com/r/Ultralytics)

## Upgrade

Make sure you are using the latest `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can run in any of these up-to-date, verified environments (with all dependencies, including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you again for your report! An Ultralytics engineer will provide further assistance soon.

**Comment by Y-T-G** [comment]
Duplicate of https://github.com/ultralytics/ultralytics/issues/17801

**Comment by Y-T-G** [comment]
If you pass `model="yolo11n.yaml"`, it will train from scratch. As long as you pass `.yaml`, it will always train from scratch. The `yolo11n.pt` checkpoint is not being downloaded for training. It's for a different reason. You can ignore that.

**Comment by Y-T-G** [comment]
> > [@Salmankm93](https://github.com/Salmankm93) What is the issue currently? The model is not being used for training. It's downloaded for AMP checks which is different from training. So you can ignore that. It will always download it for AMP checks.
> 
> Thank you a lot, I get your point. However, before, when I trained the same module, which is the yolov8 on a custom dataset from scratch, I have recognized that the GPU usage is reached to 8GB but now on the same module and the same dataset it nearly reached 4.7 GB, so is it accepted or there is an issue?.

@Salmankm93 If you want to utilize more VRAM, you can set `batch=-1`.

**Comment by Salmankm93** [comment]
@Y-T-G, if I set the batch=-1, would I have better results than 16 or 8?

**Comment by Y-T-G** [comment]
Depends. Shouldn't be a lot different.
